src/biconnected.jl

```diff
@@ -0,0 +1,188 @@
+"""
+
+biconnected Component Decomposition
+----------------------------------
+
+Any connected graph decomposes into a tree of biconnected components 
+called the block-cut tree of the graph. The blocks are attached to each 
+other at shared vertices called cut vertices or articulation points.
+
+This implementation is based on the algorithm provided by Tarjan 
+in "Depth-First Search and Linear Graph Algorithms".  
+
+
+'1' in the entry of the articulation point array denotes that the vertex is an articulation point.
+
+
+Functions
+---------
+
+biconnected_component(A::MatrixNetwork)                         
+Returns an array that represents the biconnected component number for the corrsponding 
+edge in its csr representation.
+
+enrich_biconnected(A::MatrixNetwork, biconnected_component_id::Int64)  
+Returns the largest biconnected component if the biconnected_component_id parameter is zero or
+less. Else return the component specified.  
+
+
+"""
+
+type biconnected_components_output
+    biconnected_components_label::Vector{Int64} #biconnected_component_number
+    articulation_points::Vector{Int64}  
+    A::MatrixNetwork #MatrixNetwork
+end
+
+
+
+function biconnected_component(A::MatrixNetwork)
+    n=length(A.rp)-1
+    rp=A.rp
+    ci=A.ci
+    biconnected_components_label=zeros(Int64,length(ci))
+    cn=1
+    low=zeros(Int64,n)
+    dt=-1*ones(Int64,n)
+    pred=zeros(Int64,n)
+    t=0
+    largest_bcc=0
+    edge_count_bcc=0
+    rs=Tuple{Int,Int,Int}[]
+    cs=Tuple{Int,Int,Int}[]
+    root_children= 0
+    articulation = zeros(Int64,n)
+    
+    #start dfs at 1.
+    for sv=1:n
+        v=sv
+        if dt[v]>0
+            continue
+        end
+        root_children = 0
+        low[v]=t
+        dt[v]=t;
+        t=t+1;
+        w = ci[rp[v]]
+        if (v == w) && (rp[v]+1 == rp[v+1])
+            biconnected_components_label[rp[v]]=cn
+            cn = cn+1
+            articulation[v]=1
```

Comment:
Is it possible to get the articulation points without the biconnected component map?